### ISSUE TITLE
Save ICA masses in emission rate file

### ISSUE THREAD

**Issue by twVolc** [twVolc/PyCamPermanent#233]
There's times where flow_glob fails to find a plume speed (gets inf as it things there is no lag between lines). It's then impossible to pull useful data from the sequence (technically you could look at the flow_nadeau emission rate, then divide by it's plume speed to get the ICA mass, but this is convoluted and won't work when the plume speed is calculated to be 0, which it sometimes is).

I think it would therefore be useful for the line's ICA mass to be saved in the emission rate file, so that a user could manually generate a plume speed and then get to an emission rate themselves by scaling the ICA mass.

**Comment by twVolc** [comment]
> Would the ICA mass be: `cds * pix_dists * C`?

Yes it would be the sum of that vector. So `sum(cds * pix_dists * C)`

**Comment by ubdbra001** [comment]
Okay great, that should be easy to access and add as we're passing all those values (except `C`) to `pyplis` from (e.g.) here:
https://github.com/twVolc/PyCamPermanent/blob/67c8b0f76b004e48397f3b51cef5f5b4b759c528/pycam/so2_camera_processor.py#L3311-L3312

Looks like there are multiple places `self.det_emission_rate_kgs()` is run so we'd maybe have to make changes around them all?

**Comment by twVolc** [comment]
The ICA will be the same for each flow-type, so I think for each image sequence we should only have to do this once, i.e. once for each line in the above for-loop rather than having to do this separately for each flow. So hopefully we'd only have to add this calculation to one place? Then perhaps it could be added to the `res` dictionary with a key of 'ICA' or something like that? Then hopefully this can be saved in the emission rates file

**Comment by ubdbra001** [comment]
Should it appear in all the output files or just the `flow_glob` one?

**Comment by twVolc** [comment]
Good question. It's only important that it's saved somewhere, but I guess duplicating it wouldn't be bad. I suppose if you didn't select `flow_glob` you'd miss out on ICA being saved if it were only saved in `flow_glob`, so perhaps we should duplicate it in all files.

**Comment by ubdbra001** [comment]
I think this one is almost ready, but one final question:
In `sum(cds * pix_dists) * C`, is there a possibility that `cds` or `pix_dists` would ever be `nan`s? If so, should this return a `nan` or a value ignoring the `nan` values?

**Comment by twVolc** [comment]
There's probably an ever so slight possibility a `nan` could pop up, although I think it's unlikely. To be on the safe side I'd definitely add this in as a `nansum` to ignore `nan` rather than returning `nan` in those cases.
